crates/starknet-devnet/tests/test_v3_transactions.rs

```diff
@@ -0,0 +1,472 @@
+
+        let (signer, account_address) = devnet.get_first_predeployed_account().await;
+
+        let mut account: SingleOwnerAccount<
+            &starknet_rs_providers::JsonRpcClient<starknet_rs_providers::jsonrpc::HttpTransport>,
+            starknet_rs_signers::LocalWallet,
+        > = SingleOwnerAccount::new(
+            &devnet.json_rpc_client,
+            signer,
+            account_address,
+            constants::CHAIN_ID,
+            ExecutionEncoding::New,
+        );
+        account.set_block_id(BlockId::Tag(BlockTag::Latest));
+
+        let declare_transaction =
+            account.declare_v3(Arc::new(sierra_artifact), casm_hash).send().await.unwrap();
+
+        assert_tx_successful(&declare_transaction.transaction_hash, &devnet.json_rpc_client).await;
+
+        devnet
+            .json_rpc_client
+            .get_class(BlockId::Tag(BlockTag::Latest), declare_transaction.class_hash)
+            .await
+            .unwrap();
+        let salt = Felt::from_hex_unchecked("0x123");
+        let unique = Felt::ZERO;
+        let constructor_arg = Felt::from(10);
+        let deploy_call = vec![Call {
+            to: UDC_CONTRACT_ADDRESS,
+            selector: get_selector_from_name("deployContract").unwrap(),
+            calldata: vec![
+                declare_transaction.class_hash,
+                salt,            // salt
+                unique,          // unique
+                Felt::ONE,       // constructor length
+                constructor_arg, // constructor arguments
+            ],
+        }];
+
+        let contract_address = get_udc_deployed_address(
+            salt,
+            declare_transaction.class_hash,
+            &starknet_rs_core::utils::UdcUniqueness::NotUnique,
+            &[constructor_arg],
+        );
+
+        let estimate_fee = account.execute_v3(deploy_call.clone()).estimate_fee().await.unwrap();
+        let gas_steps = estimate_fee
+            .overall_fee
+            .field_div(&NonZeroFelt::try_from(estimate_fee.gas_price).unwrap());
+        let deploy_transaction = account
+            .execute_v3(deploy_call)
+            .gas(gas_steps.to_le_digits().first().cloned().unwrap())
+            .send()
+            .await
+            .unwrap();
+        assert_tx_successful(&deploy_transaction.transaction_hash, &devnet.json_rpc_client).await;
+
+        let class_hash_of_contract = devnet
+            .json_rpc_client
+            .get_class_hash_at(BlockId::Tag(BlockTag::Latest), contract_address)
+            .await
+            .unwrap();
+
+        assert_eq!(declare_transaction.class_hash, class_hash_of_contract);
+    }
+    #[tokio::test]
+    async fn declare_from_an_account_with_insufficient_strk_tokens_balance() {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&[
+            "--gas-price-fri",
+            "10",
+            "--data-gas-price-fri",
+            "100",
```

Comment:
Why are these values used in all tests? What would happen if the defaults were used?

Reply:
[31452e8](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/625/commits/31452e82804f5f504de71c9985b1cdba2b9baf4f)